other/web-js/client/src/client-api.js

```diff
@@ -0,0 +1,212 @@
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${refresh_token}}`,
+      },
+    });
+    if (resp.status === 200) {
+      // Example 200 Response: {
+      //   "access_token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJiYWFzX2RldmljZV9pZCI6IjYzZjUxNWNhNmVmZDZkMjU1OTk5M2U2YiIsImJhYXNfZG9tYWluX2lkIjoiNWNkYjEyNDA4ZTIzMmFjNGY5NTg3ZmU4IiwiZXhwIjoxNjc3MDA5MzI0LCJpYXQiOjE2NzcwMDc1MjQsImlzcyI6IjYzZjUxYTM3ZDI0M2VmZTY1YWFhMGZmZCIsInN0aXRjaF9kZXZJZCI6IjYzZjUxNWNhNmVmZDZkMjU1OTk5M2U2YiIsInN0aXRjaF9kb21haW5JZCI6IjVjZGIxMjQwOGUyMzJhYzRmOTU4N2ZlOCIsInN1YiI6IjYzZjUxNWNhNmVmZDZkMjU1OTk5M2U1ZSIsInR5cCI6ImFjY2VzcyJ9.rbB5ZW8e88P-VKPTYNZYyCM3RKl77FuW_kBf29POr24"
+      // }
+      return await resp.json();
+    } else {
+      throw new ClientApiError(await resp.json());
+    }
+  };
+
+  emailPasswordAuth = {
+    // registerUser(credentials: { email: string, password: string }): Promise<User>
+    registerUser: async ({ email, password }) => {
+      return await this.registerUser("local-userpass", { email, password });
+    },
+    // logIn(credentials: { email: string, password: string }): Promise<User>
+    logIn: async ({ email, password }) => {
+      return await this.logIn("local-userpass", { email, password });
+    },
+  };
+}
+
+/**
+ * CredentialStorage is a lightweight wrapper around localStorage that
+ * lets users stay logged in across page refreshes. You could replace
+ * this with your own implementation if you wanted to use a different
+ * storage mechanism.
+ */
+export class CredentialStorage {
+  constructor(id) {
+    this.namespace = `@${id}`;
+  }
+  namespacedKey(key) {
+    return `${this.namespace}/${key}`;
+  }
+  get(key) {
+    const data = localStorage.getItem(this.namespacedKey(key));
+    return JSON.parse(data);
+  }
+  set(key, data) {
+    localStorage.setItem(this.namespacedKey(key), JSON.stringify(data));
+  }
+  delete(key) {
+    localStorage.removeItem(this.namespacedKey(key));
+  }
+  clearAll() {
+    localStorage.clear();
+  }
+}
+
+// A wrapper around JavaScript's built-in Error object that represents
+// errors returned by the Client API.
+// Example error: {
+//   error: 'name already in use',
+//   error_code: 'AccountNameInUse',
+//   link: 'https://realm.mongodb.com/groups/{groupId}/apps/{appId}/logs?co_id=63f506d9d243efe65aa33430'
+// }
+/**
+ * @typedef {Object} ClientApiError
+ * @property {string} name - The name of the error type: "ClientApiError"
+ * @property {string} error - A human-readable error message.
+ * @property {string} error_code - A machine-readable error code.
+ * @property {string} link - A link to a related application log in the App Services UI.
+ */
+export class ClientApiError extends Error {
+  // constructor(input: { error: string, error_code: string, link: string }): ClientApiError
+  constructor({ error, error_code, link }) {
+    super(error);
+    this.name = "ClientApiError";
+    this.error = error;
+    this.error_code = error_code;
```

Comment:
```suggestion
    this.errorCode = error_code;
```

nit: should this be camelCase like your other user-defined variables?

Reply:
I think I went with snake here to most closely match the underlying error. Doesn't matter to me though, can update.